src/views/Simulation/components/Scene/Scene.js

```diff
@@ -156,18 +158,31 @@ const Scene = () => {
       requiredUpdateStepsRef.current.all ||
       requiredUpdateStepsRef.current.highlight ||
       requiredUpdateStepsRef.current.layout ||
+      requiredUpdateStepsRef.current.paletteMode ||
       requiredUpdateStepsRef.current.render
     ) {
+      const palette = simulationTheme[theme.palette.mode];
+
       if (sceneContainerRef.current) {
         sceneContainerRef.current.removeChildren().forEach((child) => {
-          child.destroy({ children: true, texture: false, baseTexture: false });
+          child.destroy({
+            children: true,
+            texture: requiredUpdateStepsRef.current.paletteMode,
+            baseTexture: requiredUpdateStepsRef.current.paletteMode,
+          });
         });
+        if (requiredUpdateStepsRef.current.paletteMode)
+          sceneContainerRef.current.textures = generateTextures(sceneAppRef.current, palette);
       }
 
       const resetBounds = requiredUpdateStepsRef.current.all || requiredUpdateStepsRef.current.layout;
-      renderElements(sceneContainerRef, graphRef, setSelectedElementId, settings, resetBounds);
+      renderElements(sceneContainerRef, graphRef, sceneAppRef, setSelectedElementId, settings, theme, resetBounds);
       if (layoutUpdate && sceneContainerRef.current) sceneContainerRef.current.setOrigin();
-      minimapContainerRef.current?.renderElements();
+
+      if (minimapContainerRef.current) {
+        minimapContainerRef.current.createNodeTextures(palette);
```

Comment:
10 lines above, the creation of new textures is in a condition block `if (requiredUpdateStepsRef.current.paletteMode)`. Don't we need the same check here?

Also, we'll probably need to check for memory leaks after these changes